deps/MPIconstants-1.4.0/README.md

```diff
@@ -0,0 +1,43 @@
+# MPIconstants
```

Comment:
Why are we importing a local copy?

Reply:
We use `MPIconstants` to support arbitrary, external MPI libraries. This package is small, and distributing a copy seemed the easiest way.

Previously, `MPI.jl` auto-detected a few MPI implementations (essentially MPICH and OpenMPI) and then hardcoded certain constants. Other (link-time) constants were not available. `MPIconstants` extracts the constants and generates Julia code that correctly wraps C types, and generates an `.so` library that makes link-time constants available. Since this is for external, user-specified libraries, we cannot build `MPIconstants` ahead of time.

In the far future, we could ship `MPIwrapper` and use it to automatically wrap external MPI libraries, and then access everything via `MPItrampoline`. But at the moment, `MPItrampoline` is just one (of several) options.